api/spec/controllers/spree/api/line_items_controller_spec.rb

```diff
@@ -123,23 +123,23 @@ module Spree
 
         before { order.create_proposed_shipments }
 
-        it "clear out shipments on create" do
-          expect(order.reload.shipments).not_to be_empty
-          api_post :create, :line_item => { :variant_id => product.master.to_param, :quantity => 1 }
-          expect(order.reload.shipments).to be_empty
-        end
-
-        it "clear out shipments on update" do
-          expect(order.reload.shipments).not_to be_empty
-          api_put :update, :id => line_item.id, :line_item => { :quantity => 1000 }
-          expect(order.reload.shipments).to be_empty
-        end
-
-        it "clear out shipments on delete" do
-          expect(order.reload.shipments).not_to be_empty
-          api_delete :destroy, :id => line_item.id
-          expect(order.reload.shipments).to be_empty
-        end
+      #   it "clear out shipments on create" do
+      #     expect(order.reload.shipments).not_to be_empty
+      #     api_post :create, :line_item => { :variant_id => product.master.to_param, :quantity => 1 }
+      #     expect(order.reload.shipments).to be_empty
+      #   end
+
+      #   it "clear out shipments on update" do
+      #     expect(order.reload.shipments).not_to be_empty
+      #     api_put :update, :id => line_item.id, :line_item => { :quantity => 1000 }
+      #     expect(order.reload.shipments).to be_empty
+      #   end
+
+      #   it "clear out shipments on delete" do
+      #     expect(order.reload.shipments).not_to be_empty
+      #     api_delete :destroy, :id => line_item.id
+      #     expect(order.reload.shipments).to be_empty
+      #   end
```

Comment:
I'm not following. I agree on how the replace works. However these tests are not testing that the particular shipments are deleted, it's asserting that there are not any associated with the order, which is affected by this PR since Order#ensure_updated_shipments is no longer doing a `shipments.destroy_all` and waiting until the delivery state to regenerate the shipments.